sources/packages/backend/apps/api/src/services/application-change-request/application-change-request.service.ts

```diff
@@ -0,0 +1,183 @@
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import {
+  Application,
+  ApplicationEditStatus,
+  ApplicationStatus,
+  EducationProgramOffering,
+  NoteType,
+  StudentAppeal,
+  User,
+} from "@sims/sims-db";
+import { DataSource, Repository } from "typeorm";
+import { NoteSharedService, WorkflowClientService } from "@sims/services";
+import { ApplicationService } from "../application/application.service";
+import {
+  APPLICATION_NOT_FOUND,
+  INVALID_APPLICATION_EDIT_STATUS,
+} from "@sims/services/constants";
+import { CustomNamedError } from "@sims/utilities";
+
+/**
+ * Service responsible for application change request operations.
+ */
+@Injectable()
+export class ApplicationChangeRequestService {
+  constructor(
+    private readonly dataSource: DataSource,
+    @InjectRepository(Application)
+    private readonly applicationRepo: Repository<Application>,
+    private readonly noteSharedService: NoteSharedService,
+    private readonly workflowClientService: WorkflowClientService,
+    private readonly applicationService: ApplicationService,
+  ) {}
+
+  /**
+   * Assess the application change request status for the given application change request for the ministry user.
+   * @param applicationId application id for which to update the status.
+   * @param applicationEditStatus the application edit status to be updated.
+   * @param note note to be saved.
+   * @param auditUserId user that should be considered the one that is causing the changes.
+   */
+  async updateApplicationChangeRequestStatus(
+    applicationId: number,
+    applicationEditStatus:
+      | ApplicationEditStatus.ChangedWithApproval
+      | ApplicationEditStatus.ChangeDeclined,
+    note: string,
+    auditUserId: number,
+  ): Promise<void> {
+    // Get the application by id.
+    const application = await this.applicationService.getApplicationById(
```

Comment:
Why do we need to execute this query and the other one at line 108?
Can we have the query at line 108 executed instead of this one?